R/utilities-units.R

```diff
@@ -136,9 +196,47 @@ getDimensionForUnit <- function(unit) {
 getUnitsForDimension <- function(dimension) {
   validateIsString(dimension)
   dimensionTask <- getDimensionTask()
-  rClr::clrCall(dimensionTask, "AllAvailableUnitNamesForDimension", dimension)
+  rClr::clrCall(dimensionTask, "AllAvailableUnitNamesFor", dimension)
+}
+
+#' Get conversion factor from one unit to another
+#'
+#' @param fromUnit String name of the unit to convert from
+#' @param toUnit String name of the unit to convert to
+#' @param dimension Dimension of the units (conversion between mass and molar possible when
+#' \code{MW} is provided)
+#' @param MW Optional molecule weight in g/mol to use when converting, for example, from molar to mass amount or concentration
+#' @return Numerical value. A value in \code{fromUnit} multiplied by the factor
+#' corresponds to the value in \code{toUnit}
+#' @export
+getUnitConversionFactor <- function(fromUnit, toUnit, dimension, MW = NULL) {
+  validateDimension(dimension)
+
+  # Convert MW to base unit
+  if (!is.null(MW)) {
+    MW <- toBaseUnit(quantityOrDimension = Dimensions$`Molecular weight`, values = MW, unit = "g/mol")
+  }
+
+  # Conversion of fromUnit to base unit
+  convFac <- toBaseUnit(
+    quantityOrDimension = dimension, values = 1,
+    unit = fromUnit, molWeight = MW
+  )
+  # Convertion from base unit to toUnit
+  convFac <- convFac * toUnit(
+    quantityOrDimension = dimension, values = 1,
+    targetUnit = toUnit, molWeight = MW
+  )
+  return(convFac)
 }
 
+#' Supported dimensions.
+#'
+#' @include enum.R
+#'
+#' @export
+Dimensions <- list()

```

Comment:
Initialize empty and then populate after package is loaded.

Reply:
I would like not to rely on this too much for now because @abdullahhamadeh is going to implement a parsign method see #478